### ISSUE TITLE
S3: POST/PUT to bucket URLs don't route correctly on port 4566

### ISSUE THREAD

**Issue by mcintyre94** [localstack/localstack#2442]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description

We noticed this while converting a service to use Localstack for tests. The service generates S3 presigned post URLs. We're able to create and use presigned S3 URLs on port 4572 (deprecated S3 port), but not 4566, the new shared one. The same issue happens with PUT requests, which is the simplest to repro.

While just using 4572 works, this does force us to use the deprecated port and I figured it was worth opening an issue because of the discrepancy.

## Expected behavior

POST http://localhost:4566/hello (with appropriate form params) should return a 204, in the same way that POST http://localhost:4572/hello does.

PUT http://localhost:4566/hello should create a bucket and return a 200, in the same way that PUT http://localhost:4572/hello does.

## Actual behavior

Both PUT and POST http://localhost:4566/hello return a 404.

In the localstack logs:
2020-05-20T13:37:41:INFO:localstack.services.edge: Unable to find forwarding rule for host "localhost:4566", path "/hello", target header "", auth header ""

# Steps to reproduce

```bash
$ curl -i -XPUT http://localhost:4572/hello
HTTP/1.1 200 OK
Server: BaseHTTP/0.6 Python/3.8.2
Date: Wed, 20 May 2020 13:43:17 GMT
Content-Type: application/xml; charset=utf-8
content-length: 159
Access-Control-Allow-Origin: *
Last-Modified: Wed, 20 May 2020 13:43:17 GMT
x-amz-request-id: 0ABD347D7A4E0697
x-amz-id-2: MzRISOwyjmnup0ABD347D7A4E06977/JypPGXLh0OVFGcJaaO3KW/hRAqKOpIEEp
Access-Control-Allow-Methods: HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH
Access-Control-Allow-Headers: authorization,content-type,content-md5,cache-control,x-amz-content-sha256,x-amz-date,x-amz-security-token,x-amz-user-agent,x-amz-target,x-amz-acl,x-amz-version-id,x-localstack-target,x-amz-tagging
Access-Control-Expose-Headers: x-amz-version-id

<CreateBucketResponse xmlns="http://s3.amazonaws.com/doc/2006-03-01"><CreateBucketResponse><Bucket>hello</Bucket></CreateBucketResponse></CreateBucketResponse>%                                                                                                                

$ curl -i -XPUT http://localhost:4566/hello
HTTP/1.1 404 Not Found
Server: BaseHTTP/0.6 Python/3.8.2
Date: Wed, 20 May 2020 13:43:22 GMT
Content-Length: 21
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH
Access-Control-Allow-Headers: authorization,content-type,content-md5,cache-control,x-amz-content-sha256,x-amz-date,x-amz-security-token,x-amz-user-agent,x-amz-target,x-amz-acl,x-amz-version-id,x-localstack-target,x-amz-tagging
Access-Control-Expose-Headers: x-amz-version-id

{"status": "running"}%
```

## Command used to start LocalStack

`localstack start`

**Comment by alyyousuf7** [comment]
is this issue and #2430 the same?

**Comment by whummer** [comment]
Thanks for reporting this issue @mcintyre94 @alyyousuf7 . This should be fixed in #2487 - can you please give it a try with the `latest` version? Please let us know if the problem persists. Thanks

**Comment by mcintyre94** [comment]
@whummer  The issue as reported here is fixed - PUT to 4566 creates a bucket now as expected. I'm still seeing a 404 in the specific case of trying to use a presigned URL on 4566, but the repro there is a bit fiddly and currently tied a bit too much to my specific code. I'll try to find some time to make a minimal repro and verify whether there still is a localstack issue and get back to you on that one. Thanks for taking a look at this! :)
